engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/ordering/LessThanNode.java

```diff
@@ -0,0 +1,305 @@
+package org.enso.interpreter.node.expression.builtin.ordering;
+
+import com.ibm.icu.text.Normalizer;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import org.enso.interpreter.dsl.AcceptsError;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.WarningsLibrary;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+@BuiltinMethod(
+    type = "Comparable",
+    name = "less_than_builtin",
+    description = """
+        Returns true if self is less than `other`. Or throw an error if the values are
+        not comparable.
+        """
+)
+@GenerateUncached
+public abstract class LessThanNode extends Node {
+
+  public static LessThanNode build() {
+    return LessThanNodeGen.create();
+  }
+
+  public abstract Object execute(@AcceptsError Object left, @AcceptsError Object other);
+
+  @Specialization
+  boolean lessIntegers(int i, int j) {
+    return i < j;
+  }
+
+  @Specialization
+  boolean lessBools(boolean b1, boolean b2) {
+    return !b1 && b2;
+  }
+
+  @Specialization
+  boolean lessLongs(long l1, long l2) {
+    return l1 < l2;
+  }
+
+  @Specialization
+  boolean lessDoubles(double self, double other) {
+    return self < other;
+  }
+
+  @Specialization
+  boolean lessLongDouble(long self, double other) {
+    return (double) self < other;
+  }
+
+  @Specialization
+  boolean lessDoubleLong(double self, long other) {
+    return self < (double) other;
+  }
+
+  @Specialization
+  boolean lessIntLong(int self, long other) {
```

Comment:
There should be no `int` "flying" in Enso interpreter. Only `long` values. `int` shall be converted to `long` on the boundary of the Enso language.